### PR TITLE
feat: add WSK listvalues api to backend

### PR DESCRIPTION
**Pull request by akrem-chabchoub**
Implements https://github.com/kubeflow/notebooks/issues/691

New **POST**  `.../workspacekinds/{name}/podtemplate/options/listvalues` that returns **imageConfig** and **podConfig** options for that workspace kind.

Each value includes **ruleEffects** — for now always `uiHide: false` until real selector logic exists.

Work on top of: https://github.com/kubeflow/notebooks/pull/875 (addressing review comments)

### CODE REVIEW

**Review by andyatmiami — CHANGES_REQUESTED** [review]
@akrem-chabchoub 

Thanks for raising this PR - excited about the opportunities this will open up in the codebase in the future!

Got a few things I would like for you to take a look at to _harden_ this good foundation you have laid.  Let me know if you have any questions.

Once you get these changes in - I'll focus my subsequent review on testing within our `Tilt` environment to assert appropriate API behavior 💯

File: workspaces/backend/internal/models/workspacekinds/types.go [hunk 1]
```diff
@@ -109,3 +115,89 @@ const (
...
+type ListValuesRequest struct {
+	Context *ListValuesContext `json:"context,omitempty"`
+}
+
+// Validate validates the listvalues request data by delegating to Context when present.
+func (d *ListValuesRequest) Validate(prefix *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	if d == nil {
+		return errs
+	}
```
Comment (andyatmiami) [inline workspaces/backend/internal/models/workspacekinds/types.go]: I don't think we need to have a `nil` guard here as this is the top level `Data` object which we check for nil-ness directly within the handler.

additionally, our other `Validate` functions don't do a nil-ness check on the receiver type.

File: workspaces/backend/internal/models/workspacekinds/types.go [hunk 2]
```diff
@@ -109,3 +115,89 @@ const (
...
 	RedirectMessageLevelDanger  RedirectMessageLevel = "Danger"
 )
+
+type ListValuesRequest struct {
+	Context *ListValuesContext `json:"context,omitempty"`
+}
+
+// Validate validates the listvalues request data by delegating to Context when present.
+func (d *ListValuesRequest) Validate(prefix *field.Path) field.ErrorList {
+	var errs field.ErrorList
```
Comment (andyatmiami) [inline workspaces/backend/internal/models/workspacekinds/types.go]: To be consistent with other `Validate` functions - I think we want to return a type of `[]*field.Error` - which then also affects the implementation in the body..

File: workspaces/backend/internal/models/workspacekinds/types.go [hunk 3]
```diff
@@ -109,3 +115,89 @@ const (
...
+
+type ListValuesContext struct {
+	// TODO: Namespace is reserved for future use, no behavior implemented yet.
+	Namespace   *ContextNamespace   `json:"namespace,omitempty"`
+	PodConfig   *ContextPodConfig   `json:"podConfig,omitempty"`
+	ImageConfig *ContextImageConfig `json:"imageConfig,omitempty"`
+}
+
+// Validate validates the context fields when present.
+func (c *ListValuesContext) Validate(prefix *field.Path) field.ErrorList {
```
Comment (andyatmiami) [inline workspaces/backend/internal/models/workspacekinds/types.go]: To be consistent with other `Validate` functions - I think we want to return a type of `[]*field.Error` - which then also affects the implementation in the body..

File: workspaces/backend/internal/models/workspacekinds/types.go [hunk 4]
```diff
@@ -109,3 +115,89 @@ const (
...
+	var errs field.ErrorList
+	if c == nil {
+		return errs
+	}
```
Comment (andyatmiami) [inline workspaces/backend/internal/models/workspacekinds/types.go]: (just commenting "out loud" - no change requested here)

This `nil` check is OK if we are wishing to allow users to call this endpoint without the `context:{}` object... so it seems fine to leave to me

**Review by andyatmiami — CHANGES_REQUESTED** [review]
PR is looking really good... just a few fairly trivial comments to keep alignment with existing code practices...

Exciting to see the API really start to take shape!

Image

File: workspaces/backend/internal/models/workspacekinds/types.go (hunk 1 above)
Comment (akrem-chabchoub) [reply workspaces/backend/internal/models/workspacekinds/types.go]: Yes, my bad, I updated it

File: workspaces/backend/internal/models/workspacekinds/types.go (hunk 4 above)
Comment (akrem-chabchoub) [reply workspaces/backend/internal/models/workspacekinds/types.go]: Thanks, I’ll keep it as-is.

**Comment by akrem-chabchoub** [comment]
Thanks @andyatmiami for the review 💯 
I addressed all of your comments 🚀

**Comment by thesuperzapper** [comment]
@akrem-chabchoub as discussed on call, I have rebased and pushed a commit https://github.com/kubeflow/notebooks/pull/972/commits/4ad5e3d6756cc31b2ec22f7e66104b2315f7fdce on top of the branch to fix the issues we discussed.

**Review by andyatmiami — APPROVED** [review]
/lgtm
/approve

thanks for this work @akrem-chabchoub and appreciate the thorough review to align/re-enforce conventions @thesuperzapper 

agree with the changes delivered and also tested this locally in kind using `Bruno` API client to verify positive and negative scenarios.

all tests check out - lets get this merged! 💯
